src/Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore/Internal/DiagnosticsEntityFrameworkCoreLoggerExtensions.cs

```diff
@@ -0,0 +1,169 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore.Internal
+{
+    internal static class DiagnosticsEntityFrameworkCoreLoggerExtensions
+    {
+        private static readonly Action<ILogger, Exception> _noContextType;
+        private static readonly Action<ILogger, Type, Exception> _invalidContextType;
+        private static readonly Action<ILogger, string, Exception> _contextNotRegistered;
+        private static readonly Action<ILogger, string, Exception> _requestPathMatched;
+        private static readonly Action<ILogger, string, Exception> _applyingMigrations;
+        private static readonly Action<ILogger, string, Exception> _migrationsApplied;
+        private static readonly Action<ILogger, string, Exception, Exception> _migrationsEndPointMiddlewareException;
+        private static readonly Action<ILogger, Type, Exception> _attemptingToMatchException;
+        private static readonly Action<ILogger, string, Exception> _contextNotRegisteredDatabaseErrorPageMiddleware;
+        private static readonly Action<ILogger, Exception> _notRelationalDatabase;
+        private static readonly Action<ILogger, Exception> _noRecordedException;
+        private static readonly Action<ILogger, Exception> _noMatch;
+        private static readonly Action<ILogger, Exception> _matched;
+        private static readonly Action<ILogger, Exception, Exception> _databaseErrorPageMiddlewareException;
+
+        static DiagnosticsEntityFrameworkCoreLoggerExtensions()
+        {
+            _noContextType = LoggerMessage.Define(
+                LogLevel.Error,
+                1,
+                "No context type was specified. Ensure the form data from the request includes a contextTypeName value, specifying the context to apply migrations for.");
+
+            _invalidContextType = LoggerMessage.Define<Type>(
+                LogLevel.Error,
+                2,
+                "The context type '{ContextType}' could not be loaded. Ensure this is the correct type name for the context you are trying to apply migrations for.");
+
+            _contextNotRegistered = LoggerMessage.Define<string>(
+                LogLevel.Error,
+                3,
+                "The context type '{Context}' was not found in services. This usually means the context was not registered in services during startup. You probably want to call AddScoped<>() inside the UseServices(...) call in your application startup code.");
```

Comment:
`'{ContextType}'`

Reply:
cc @Eilon These error messages are the ones I was referring to. Maybe they can be reworded